script/support/sample_move.rb

```diff
@@ -0,0 +1,296 @@
+          .map(&:id)
+      )
+      sample
+        .aliquots
+        .where(study_id: study_from_id)
+        .find_each do |aliquot|
+          aliquot.update!(study_id: study_to_id)
+          aliquot.receptacle.tap do |asset|
+            if tube_classes.include?(asset.labware.class.name)
+              update_create_requests_on(asset)
+              comment_on.call(asset)
+              tubes << asset
+            elsif asset.is_a?(Well)
+              update_create_requests_on(asset)
+              comment_on.call(asset)
+              stock_wells << asset if asset.plate.stock_plate?
+            elsif asset.is_a?(Lane)
+              lane_ids << asset.id
+            elsif asset.is_a?(PacBioLibraryTube)
+              pb_tube_ids << asset.id
+            end
+          end
+        end
+
+      sample.study_samples.find_each do |study_sample|
+        study_sample.update!(study_id: study_to_id)
+        study_links = StudySample.where(sample_id: study_sample.sample_id, study_id: study_from_id)
+        study_links.each do |link|
+          puts "Sample already associated with #{study_to_id} => Destroying #{link.inspect}\n"
+          link.destroy
+        end
+      end
+      puts 'Finished saving study_sample'
+    end
+
+    raise 'Hell!!!... in test mode' if mode == 'test'
+    unless stock_wells.compact.empty?
+      puts 'Rebroadcasting well stock resource messages.. '
+      stock_wells.uniq.each do |sw|
+        sw.touch #correct behaviour
+        sw.messengers.first.resend
+      end
+    end
+
+    unless tubes.empty?
+      puts 'Rebroadcasting tube stock resource messages.. '
+      tubes.each do |tube|
+        tube.touch #correct behaviour
+        tube.messengers.first.resend
+      end
+    end
+
+    unless lane_ids.empty?
+      puts 'Rebroadcasting batches...'
+      lanes = Lane.where(id: lane_ids.uniq)
+      update_seq_requests(lanes.map(&:requests_as_target).flatten)
+      lanes
+        .map(&:creation_batches)
+        .flatten
+        .uniq
+        .each do |batch|
+          puts "lane batches: #{batch.id}"
+          batch.touch #correct behaviour
+        end
+    end
+
+    unless pb_tube_ids.empty?
+      requests = PacBioLibraryTube.where(id: pb_tube_ids.uniq).map(&:requests).flatten
+      update_seq_requests(requests)
+      requests
+        .filter_map(&:batch)
+        .uniq
+        .each do |batch|
+          puts "pb batches: #{batch.id}"
+          batch.touch #correct behaviour
+        end
+    end
+
+    unless fluidigm_plates.empty?
+      puts "Updating\n#{fluidigm_plates.uniq.inspect}"
+      update_fluidigm_plates(fluidigm_plates.uniq)
+    end
+  end
+end
+# rubocop:enable Rails/SkipsModelValidations
+# rubocop:enable Metrics/PerceivedComplexity
+# rubocop:enable Metrics/MethodLength
+# rubocop:enable Metrics/AbcSize
+# rubocop:enable Metrics/CyclomaticComplexity
+# rubocop:enable Layout/LineLength
+# rubocop:enable Metrics/ParameterLists
+# rubocop:enable Metrics/BlockLength
```

Comment:
If these must stay in this file (see other comment above) I think I'd prefer to see some symmetry in when they're disabled and when they're re-enabled.  At the moment they're turned off sporadically in the code and then all turned on at the end.  It would be better, in my mind, to disable them as the first thing in this file if they will be turned on all together at the end of it.  It will make it easier to keep track of what's off and back on in the future.